deployer/deployer.js

```diff
@@ -74,12 +74,18 @@ class Deployer {
 	 * @param {*} contract the contract to be deployed
 	 * @param {*} deploymentArguments the deployment arguments
 	 */
-	async _preValidateArguments(contract, deploymentArguments) {
+	async _preValidateArguments(contract, libraries, deploymentArguments) {
 		if (!(isValidContract(contract))) {
 			await this._logAction(this.constructor.name, contract.contractName, '', 1, '-', '-', 'Invalid contract object');
 			throw new Error(`Passed contract is not a valid contract object. It needs to have bytecode, abi and contractName properties`);
 		}
 
+		if (isValidLibrary(libraries)) {
```

Comment:
I thing if you rethink this you can write this much better. This if is of no use here